agent/pkg/chassis/protocol/tcp/handler.go

```diff
@@ -76,7 +76,13 @@ func (h *L4ProxyHandler) Handle(chain *handler.Chain, i *invocation.Invocation,
 			r.Err = fmt.Errorf("l4 proxy dial error: %v", err)
 			return
 		}
-		defer rconn.Close()
+		closeOnce := &sync.Once{}
+		defer func() {
+			closeOnce.Do(func() {
+				lconn.Close()
+				rconn.Close()
+			})
+		}()
```

Comment:
this code logic already implemented in function pipe()

Reply:
rconn may close twice if not protected using the one `sync.Once`